libbeat/cloudid/cloudid.go

```diff
@@ -150,8 +150,8 @@ func decodeCloudID(cloudID string) (string, string, error) {
 	return esURL.String(), kibanaURL.String(), nil
 }
 
-// decodeCloudAuth splits the cloud.auth into username and password.
-func decodeCloudAuth(cloudAuth string) (string, string, error) {
+// DecodeCloudAuth splits the cloud.auth into username and password.
+func DecodeCloudAuth(cloudAuth string) (string, string, error) {
```

Comment:
Hm... I guess these details are used for Elasticsearch and Kibana? We might want to add those to the struct as well?

alternatively one can hide the parsing behind a structure (lazy or eager parsing is possible)

```
type CloudID struct {
  id string
  username, password, esURL, kibURL string
}

func NewCloudID(id string) (*CloudID, error) {
  // parse id and fill structure
}

func (cid *CloudID) ID() string {
  return cid
}

func (cid *CloudID) Username() string {
  if cid.username == "" {
    cid.parseCredentials()
  }
  return cid.username
}

...
```